src/libraries/Native/Unix/System.Security.Cryptography.Native.Android/pal_rsa.c

```diff
@@ -153,18 +158,24 @@ PALEXPORT RSA* CryptoNative_DecodeRsaPublicKey(uint8_t* buf, int32_t len)
 
     if (CheckJNIExceptions(env))
     {
-        CryptoNative_RsaDestroy(rsa);
+        AndroidCryptoNative_RsaDestroy(rsa);
         return FAIL;
     }
 
     return rsa;
 }
 
-PALEXPORT int32_t CryptoNative_RsaSignPrimitive(int32_t flen, uint8_t* from, uint8_t* to, RSA* rsa)
+PALEXPORT int32_t AndroidCryptoNative_RsaSignPrimitive(int32_t flen, uint8_t* from, uint8_t* to, RSA* rsa)
 {
     if (!rsa)
         return RSA_FAIL;
 
+    if (!rsa->privateKey)
+    {
+        LOG_ERROR("RSA private key required to sign.");
```

Comment:
Are these put into exception messages, or just useful for debugging the shim?  Mainly I ask because the decryption primitive didn't log, but the signature primitive did.

Reply:
These are logged into the Android device logs for the app. This is where the Java exception messages and traces are dumped as well. I believe they are accessible in the app logs to developers.